File: app/src/main/java/com/hjq/demo/manager/MmkvUtil.java
package com.hjq.demo.manager;

import com.tencent.mmkv.MMKV;

/**
 * 腾讯文件存值 工具 2021年3月4日17:40:37
 */
public class MmkvUtil {
    private static MMKV kv;
    public static String Hosts = "hosts";
    public static String HostsIndex = "hosts_index";
    public static String Port = "key0";
    public static String LastReq = "last_req_time";
    public static String Baudrate = "key1";
    //    public static String DeviceCode = "dev_code";
    public static String Token = "token";
    public static String Version = "version";
    public static String MN = "machine-no";
    public static String OutMp3 = "good_out_mp3";
    static public String DeveloperOpenDebug = "k9";
    static public String AdUrl = "ad_url";
    static public String AdLocalfile = "ad_file";

    static public String BUILD_TYPE = "BUILD_TYPE";
    static public String DEBUG = "DEBUG";
    static public String APPLICATION_ID = "APPLICATION_ID";
    static public String BUGLY_ID = "BUGLY_ID";
    static public String VERSION_CODE = "VERSION_CODE";
    static public String VERSION_NAME = "VERSION_NAME";
    static public String HOST_URL = "HOST_URL";


    private static MMKV init() {
        if (kv == null) {
            kv = MMKV.defaultMMKV();
        }
        return kv;

    }

    public static boolean save(String key, boolean value) {
        init();
        return kv.encode(key, value);
    }

    public static boolean save(String key, int value) {
        init();
        return kv.encode(key, value);
    }

    public static boolean save(String key, long value) {
        init();
        return kv.encode(key, value);
    }

    public static boolean save(String key, double value) {
        init();
        return kv.encode(key, value);
    }

    public static boolean save(String key, float value) {
        init();
        return kv.encode(key, value);
    }

    public static boolean save(String key, String value) {
        init();
        return kv.encode(key, value);
    }

    public static boolean save(String key, byte[] value) {
        init();
        return kv.encode(key, value);
    }

    public static Boolean getBool(String key) {
        init();
        return kv.decodeBool(key);
    }

    public static byte[] getBytes(String key) {
        init();
        return kv.decodeBytes(key);
    }

    public static double getDouble(String key, double defaultValue) {
        init();
        return kv.decodeDouble(key, defaultValue);

    }

    public static int getInt(String key, int defaultValue) {
        init();
        return kv.decodeInt(key, defaultValue);
    }

    public static long getLong(String key, long defaultValue) {
        init();
        return kv.decodeLong(key, defaultValue);
    }

    public static String getString(String key, String defaultValue) {
        init();
        return kv.decodeString(key, defaultValue);
    }
}

File: app/src/main/java/com/hjq/demo/manager/MmkvUtil.kt
package com.hjq.demo.manager

import com.tencent.mmkv.MMKV

/**
 * 腾讯文件存值 工具 2021年3月4日17:40:37
 */
object MmkvUtil {
    private var kv: MMKV? = null

    const val Hosts = "hosts"
    const val HostsIndex = "hosts_index"
    const val Port = "key0"
    const val LastReq = "last_req_time"
    const val Baudrate = "key1"
    const val Databits = "Databits"
    const val Parity = "Parity"
    const val Stopbits = "Stopbits"
    const val Flowcon = "Flowcon"
    const val Token = "token"
    const val Version = "version"
    const val MN = "machine-no"
    const val OutMp3 = "good_out_mp3"
    const val DeveloperOpenDebug = "k9"
    const val AdUrl = "ad_url"
    const val AdLocalfile = "ad_file"

    const val BUILD_TYPE = "BUILD_TYPE"
    const val DEBUG = "DEBUG"
    const val APPLICATION_ID = "APPLICATION_ID"
    const val BUGLY_ID = "BUGLY_ID"
    const val VERSION_CODE = "VERSION_CODE"
    const val VERSION_NAME = "VERSION_NAME"
    const val HOST_URL = "HOST_URL"

    private fun init(): MMKV {
        return kv ?: run {
            val default = MMKV.defaultMMKV()
            kv = default
            default
        }
    }

    fun save(key: String, value: Boolean): Boolean {
        return init().encode(key, value)
    }

    fun save(key: String, value: Int): Boolean {
        return init().encode(key, value)
    }

    fun save(key: String, value: Long): Boolean {
        return init().encode(key, value)
    }

    fun save(key: String, value: Double): Boolean {
        return init().encode(key, value)
    }

    fun save(key: String, value: Float): Boolean {
        return init().encode(key, value)
    }

    fun save(key: String, value: String?): Boolean {
        return init().encode(key, value)
    }

    fun save(key: String, value: ByteArray?): Boolean {
        return init().encode(key, value)
    }

    fun getBool(key: String): Boolean {
        return init().decodeBool(key)
    }

    fun getBytes(key: String): ByteArray? {
        return init().decodeBytes(key)
    }

    fun getDouble(key: String, defaultValue: Double): Double {
        return init().decodeDouble(key, defaultValue)
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return init().decodeInt(key, defaultValue)
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return init().decodeLong(key, defaultValue)
    }

    fun getString(key: String, defaultValue: String?): String? {
        return init().decodeString(key, defaultValue)
    }
}
